lib/nosedrum/storage.ex

```diff
@@ -1,75 +1,142 @@
 defmodule Nosedrum.Storage do
   @moduledoc """
-  Storages contain commands and are used by command invokers to look up commands.
+  `Storage`s keep track of your Application Command names and their associated modules. A `Storage` handles incoming
+  `t:Nostrum.Struct.Interaction.t/0`s, invoking `c:Nosedrum.ApplicationCommand.command/1` callbacks
+  and responding to the Interaction.
 
-  How you start a storage is up to the module itself - what is
-  expected is that storage modules implement the behaviours
-  documented in this module.
-
-  The public-facing API of storage modules takes an optional argument,
-  the storage process or other information used to identify the storage
-  such as an ETS table name.
+  In addition to tracking commands locally for the bot, a `Storage` is
+  responsible for registering an Application Command with Discord when `c:add_command/4`
+  or `c:remove_command/4` is called.
   """
+  @moduledoc since: "0.4.0"
+  alias Nostrum.Struct.{Guild, Interaction}
 
-  @typedoc """
-  A single command module or mapping of subcommand names to command groups.
+  @callback_type_map %{
+    pong: 1,
+    channel_message_with_source: 4,
+    deferred_channel_message_with_source: 5,
+    deferred_update_message: 6,
+    update_message: 7
+  }
 
-  In addition to subcommand names, the key `:default` can be specified by
-  the module. `:default` should be invoked when none of the subcommands in the
-  map match.
-  """
-  @type command_group ::
-          module
-          | %{optional(:default) => command_group, required(String.t()) => command_group}
+  @flag_map %{
+    ephemeral?: 64
+  }
+
+  @type command_scope :: :global | Guild.id() | [Guild.id()]
 
   @typedoc """
-  The "invocation path" of the command.
+  Defines a structure of commands, subcommands, subcommand groups, as
```

Comment:
This should also be on two separate lines. And maybe the link should be at the end, in a `## References` section?